scmail.py

```diff
@@ -0,0 +1,259 @@
+        print(f'The {config[i]} is: {config[i+1]}')
+
+    print()
+    logging.info('Config file exists. Load Successful.')
+    return config[1], config[3]
+
+
+@click.pass_context
+def set_user_info(ctx, gnupghome, email):
+    """Set user information and return gnupghome and email."""
+    config = 'gnupghome=' + gnupghome + '\n' + 'email=' + email
+
+    # Store config file and other information.
+    ctx.config.touch(exist_ok=False)
+    with ctx.config.open('w') as f:
+        f.write(config)
+
+    logging.info('Setting user information finish.')
+    return gnupghome, email
+
+
+# @click.command()
+# @click.option('--gnupghome', prompt='Enter your gnupg home dir', default='gnupgkeys',
+#               help='The home dir of gnupg key pairs.')
+# @click.option('--email', prompt='Enter your email', default=getlogin() + '@scmail.dev', help='The email of user.')
+# @click.pass_context
+# def update_user_info(ctx, gnupghome, email):
+#     """Update gnupg home dir and personal email"""
+#     pass
+
+
+@click.command()
+@click.option('--key-type', prompt='Enter key type', default="RSA", help='The algorithms to generate the key. ex. RSA')
+@click.option('--key-length', prompt='Enter key length', default=1024, help='The length of the key.')
+@click.option('--expire-date', prompt='Enter expire date', default="2y", help='The expire time of the key pair. 0 or empty for never expire.')
+@click.password_option('--password', prompt='Enter private key password', help='Password of private key.')
+@click.pass_context
+def create_key(ctx, key_type, key_length, expire_date, password):
+    """Create gnupg key pairs."""
+    logging.debug(f'Information about key:\nkey type: {key_type}\nkey length: {key_length}\nexpire date: {expire_date}\n')
+
+    # Warning if key never expire and user want to continue.
+    if expire_date == 0 and not click.confirm('0 means never expire, Do you want to continue?'):
+        logging.warning('Never expire key will be created.')
+
+    key = ctx.parent.gpg.create(password, key_type=key_type, key_length=key_length, expire_date=expire_date)
+    ctx.parent.pp.pprint(key.__dict__)
+    logging.info('Key Creation finished.')
+
+
+@click.command()
+@click.option('--show-private', prompt='Do you want to show the private keys?', default=False, is_flag=True,
+              help='Whether show the private keys')
+@click.pass_context
+def list_keys(ctx, show_private):
+    """List and print all the keys in the gnupg home dir."""
+    keys = ctx.parent.gpg.list_keys(show_private)
+    key_type = 'public' if show_private is False else 'private'
+
+    logging.info(f'{len(keys)} {key_type} keys exist.')
+    ctx.parent.pp.pprint(keys.__dict__)
+    logging.info('List keys finished.')
+
+
+@click.command()
+@click.pass_context
+def register(ctx):
+    """Register sc_mail API."""
+    # Request register.
+    logging.debug('begin register.')
+    fingerprint = ctx.parent.gpg.list_keys(True).curkey.get('fingerprint')
```

Comment:
Ran into an error with this line. Perhaps some error checking is needed here?

Reply:
Got it, thanks.